<!doctype html>
<html>

    <head>

        <title>GRNmap - Publications</title>
        <link rel="stylesheet" type="text/css" href="http://dondi.github.io/GRNsight/assets/css/bootstrap.min.css">
        <link rel="stylesheet" type="text/css" href="http://dondi.github.io/GRNsight/assets/css/main.css">

        <script src="https://code.jquery.com/jquery.js"></script>
        <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
        <script type='text/javascript' src='js/main.js'></script>
        <meta charset='utf-8'>

        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

            ga('create', 'UA-59073426-1', 'auto');
            ga('send', 'pageview');

        </script>

    </head>

    <body>

        <div id='header'>
            <div id='logo'>
                <!--a href="http://dondi.github.io/GRNsight"><img id='GRNsightLogo' src='images/GRNsight_logo_20140710_main_resized.jpg' alt='GRNsight' width='200' height='118'--></a></div>
            <div id='lmu'>
                <a href="http://www.lmu.edu" target='_blank'><img src='images/LMULogoHQ2.png' alt='LMU logo' width='200' height='118' id='lmuLogo' /></a><br /><br /><br />
                <p id='subtextLMU'>A Project of the<br />LMU Biomathematics Group</p>
            </div>
            <p id='title'><a href="http://kdahlquist.github.io/GRNmap">GRNmap</a></p>
            <p id='subtext'>Gene Regulatory Network modeling and parameter estimation.</p>
        </div>

        <div id='navBar' class='.backBox'>
            <div id='stickyBar' data-spy='affix' data-offset-top='140'>
                <ul class='nav'>
                    <li><a href='index.html'>Home</a></li>
                    <li><a href='about.html'>About</a></li>
                    <li><a href='news.html'>News</a></li>
                    <li><a href='documentation.html'>Documentation</a></li>
                    <li><a href='downloads.html'>Downloads</a></li>
                    <li><a href='https://github.com/kdahlquist/GRNmap/wiki' target="_blank">Wiki</a></li>
                    <li><a href='https://github.com/kdahlquist/GRNmap/' target="_blank">Developers</a></li>
                    <li><a href='publications.html'>Publications</a></li>
                    <li><a href='people.html'>People</a></li>
                    <li><a href='contact.html'>Contact</a></li>
                    <li><a href='links.html'>Links</a></li>
                </ul>
                <p id='modText' class='modNav'>
                    Last modified: 5/19/16
                </p>
            </div>
        </div>

        <div id='mainContent'>
            <p id='pageTitle'>Publications</p>
            <div id='pageContent'>
                
                <div id="accordian" class="panel-group">
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <p class="panel-title aboutHeaders">
                                <a href="#papers" data-toggle="collapse">
                                    Papers
                                </a>
                            </p>
                        </div><!-- panel-heading -->
                        <div id="papers" class="panel-collapse collapse in">
                            <div class="panel-body">
                                <p class="abstract">
                                <span style='font-weight:bold'>September 29, 2015</span><br>
                                
                                Dahlquist, K.D., Fitzpatrick, B.G., Camacho, E.T., Entzminger, S.D., and Wanner, N.C. (2015) Parameter Estimation for Gene Regulatory Networks from Microarray Data: Cold Shock Response in Saccharomyces cerevisiae. <i>Bulletin of Mathematical Biology</i>, 77(8), 1457-1492, published online September 29, 2015. DOI: 10.1007/s11538-015-0092-6. 
                                <br>
                                <br>
                                <a href="http://link.springer.com/article/10.1007/s11538-015-0092-6/fulltext.html" target="_blank">View full text in HTML </a><br>
                                <a href="http://link.springer.com/content/pdf/10.1007%2Fs11538-015-0092-6.pdf" target="_blank">Download PDF (1205 KB)</a><br> 
                                <a href="http://link.springer.com/content/esm/art:10.1007/s11538-015-0092-6/file/MediaObjects/11538_2015_92_MOESM1_ESM.zip" target="_blank">Download electronic supplementary material (65 KB)</a><br><br>

                               
                                    <span style="font-weight:bold">Abstract</span><br>
                                
                                    We investigated the dynamics of a gene regulatory network controlling the cold shock response in budding yeast, Saccharomyces cerevisiae. The medium-scale network, derived from published genome-wide location data, consists of 21 transcription factors that regulate one another through 31 directed edges. The expression levels of the individual transcription factors were modeled using mass balance ordinary differential equations with a sigmoidal production function. Each equation includes a production rate, a degradation rate, weights that denote the magnitude and type of influence of the connected transcription factors (activation or repression), and a threshold of expression. The inverse problem of determining model parameters from observed data is our primary interest. We fit the differential equation model to published microarray data using a penalized nonlinear least squares approach. Model predictions fit the experimental data well, within the 95 % confidence interval. Tests of the model using randomized initial guesses and model-generated data also lend confidence to the fit. The results have revealed activation and repression relationships between the transcription factors. Sensitivity analysis indicates that the model is most sensitive to changes in the production rate parameters, weights, and thresholds of Yap1, Rox1, and Yap6, which form a densely connected core in the network. The modeling results newly suggest that Rap1, Fhl1, Msn4, Rph1, and Hsf1 play an important role in regulating the early response to cold shock in yeast. Our results demonstrate that estimation for a large number of parameters can be successfully performed for nonlinear dynamic gene regulatory networks using sparse, noisy microarray data.
                                </p>
                            </div><!-- panel-body -->
                        </div><!-- panel-collapse collapse in-->
                    </div><!-- panel panel-default -->
                    
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <p class="panel-title aboutHeaders">
                                <a href="#presentations" data-toggle="collapse">
                                    Presentations
                                </a>
                            </p>
                        </div><!-- panel-heading -->
                        <div id="presentations" class="panel-collapse collapse">
                            <div class="panel-body">
                                <p class="abstract">
                                    <span style="font-weight:bold"><a href="http://experimentalbiology.org/PDFS/2016/EB-2016-Program-Web-FullProgram.aspx">Experimental Biology 2016, San Diego, CA</a></span><br><br>
                                    April 4, 2016 <br>
                                    <a href="http://www.fasebj.org/content/30/1_Supplement/819.16.abstract?sid=a80e9e6c-998f-4889-b1fd-ddcc75ab8bda">GRNmap and GRNsight: open source software for dynamical systems modeling and visualization of medium-scale gene regulatory networks</a> <br>
                                    <span style="font-weight:bold">Kam D. Dahlquist</span>, Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Monica V. Hong, Kristen M. Horstmann, Kayla C. Jackson, K. Grace Johnson, Tessa A. Morris, Trixie Anne M. Roque, Mihir Samdarshi, Anindita Varshneya, Natalie E. Williams, and Kevin W. Wyllie<br><br>
                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Our group has developed a MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach (Dahlquist et al. 2015, DOI: 10.1007/s11538-015-0092-6) to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. GRNmap has options for using a sigmoidal or Michaelis-Menten production function. Parameters for a series of related networks, ranging in size from 15 to 35 genes, were optimized against DNA microarray data measuring the transcriptional response to cold shock in wild type and five strains individually deleted for the transcription factors, Cin5, Gln3, Hap4, Hmo1, Zap1, of budding yeast, Saccharomyces cerevisiae BY4741. Model predictions fit the experimental data well, within the 95% confidence interval. Open source code and a compiled executable that can run without a MATLAB license are available from <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>. GRNsight is an open source web application for visualizing such models of gene regulatory networks. GRNsight accepts GRNmap- or user-generated spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph of the GRN model. The application colors the edges and adjusts their thicknesses based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users can then modify the graph to define the best visual layout for the network. The GRNsight open source code and application are available from <a href="http://dondi.github.io/GRNsight/index.html">http://dondi.github.io/GRNsight/index.html</a>.
                                </p>
                                <div class="abstract">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/key/NC3auEnK2HuL1f" width="595" height="485" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen> </iframe> <div style="margin-bottom:5px"> <strong> <a href="//www.slideshare.net/GRNsight/dahlquist-experimental-biology20160404" title="Dahlquist experimental biology_20160404" target="_blank">Presentation slides</a> </strong> from <strong><a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a></strong> on SlideShare</div>   
                </div>
<br>
                                <p class="abstract">
                                    <span style="font-weight:bold"><a href="http://ccbs.uci.edu/scsb2015home">Southern California Systems Biology Conference</a></span><br><br>
                                    January 31, 2015 <br>
                                    GRNmap and GRNsight: Open Source Software for Dynamical Systems Modeling and Visualization of Medium-Scale Gene Regulatory Networks<br>
                                    <span style="font-weight:bold">Kam D. Dahlquist</span>,  Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Nicholas A. Rohacz, Katrina Sherbina, Britain J. Southwick, and Anindita Varshneya <br><br>
                                Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Our group has developed a MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. Parameters for a 21-gene network were optimized against DNA microarray data measuring the transcriptional response to cold shock in wild type and four transcription factor deletion strains of budding yeast, Saccharomyces cerevisiae. Model predictions fit experimental data well, within the 95% confidence interval. Open source code and a compiled executable that can run without a MATLAB license are available from <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>. GRNsight is an open source web application for visualizing such models of gene regulatory networks. GRNsight accepts GRNmap- or user-generated spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph of the GRN model. The application colors the edges and adjusts their thicknesses based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users can then modify the graph to define the best visual layout for the network. The GRNsight code and application are available from <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/index.html</a>. This work was partially supported by NSF award 0921038.

                                </p>
                                <div class="abstract">
                                    <iframe src="http://www.slideshare.net/slideshow/embed_code/44236648" width="425" height="355" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen> </iframe> 
                                    <div style="margin-bottom:5px"> 
                                        <a href="http://www.slideshare.net/GRNsight/dahlquist-so-calsysbio20140131" title="Dahlquist so calsysbio_20140131" target="_blank">Presentation Slides</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a>
                                    </div>
                                </div>
                            </div><!-- panel-body -->
                        </div><!-- panel-collapse collapse-->
                    </div><!-- panel panel-default -->
                    
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <p class="panel-title aboutHeaders">
                                <a href="#posters" data-toggle="collapse">
                                    Posters
                                </a>
                            </p>
                        </div><!-- panel-heading -->
                        <div id="posters" class="panel-collapse collapse">
                            <div class="panel-body">
                                <p class='abstract'>
                                    <span style='font-weight:bold'>Loyola Marymount Undergraduate Research Symposium</span><br><br>
                                    March 21, 2015 <br>
                                    Software Refactoring and Usability Enhancement for GRNmap, a Gene Regulatory Network Modeling Application<br>
                                    Juan Carrillo, Trixie Anne Roque <br>
                                    Dr. Kam D. Dahlquist, Dr. Ben G. Fitzpatrick <br><br>
                                    A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. The dynamics of a GRN is how gene expression in the network changes over time. Over a period of several years, our group has developed a complex MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs from budding yeast, Saccharomyces cerevisiae. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data, and then performs a forward simulation of the dynamics of the network. The large number of developers and time span of development led to a code base that was difficult to revise and adjust. We therefore refactored the script-based software with global variables into a function-based package that uses an object to carry relevant information from function to function. This modular approach allows for cleaner, less ambiguous code and increased maintainability. We then used the MATLAB compiler to create an executable file that can be run on any Windows machine without the need of a MATLAB license, increasing the accessibility of our program. The GRNmap code and executable is available from <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a> under an open source license. We have now adopted a test-driven development framework that will speed debugging and ensure that future modifications to the code preserve correct functionality.<br><br>
                                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/Carrillo_Roque_LMU_Symposium_2015.pptx?raw=true'>here</a>.
                                </p>
                                <br>
                                
                                <p class='abstract'>
                                    <span style='font-weight:bold'>SACNAS  (Society for the Advancement of Chicanos and Native Americans in Science) National Conference</span> <br><br>
                                    October 16-18, 2014<br>
                                    Software Refactoring and Usability Enhancement for GRNmap, a Gene Regulatory Network Modeling Application <br>
                                    Juan Carrillo, Katrina Sherbina <br>
                                    Dr. Kam D. Dahlquist, Dr. Ben G. Fitzpatrick <br><br>
                                    A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. The dynamics of a GRN is how gene expression in the network changes over time. Over a period of several years, our group has developed a complex MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of a 21-transcription factor GRN from budding yeast, Saccharomyces cerevisiae. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data, and then performs a forward simulation of the dynamics of the network. The large number of developers and time span of development led to a code base that is difficult to revise and adjust. We therefore refactored the script-based software with global variables into a function-based package that uses an object to carry relevant information from function to function. This modular approach allows for cleaner, less ambiguous code and increased maintainability. Further revisions to the model will also be easier to implement. In addition, we have added a simple user interface, removing the need for users to edit MATLAB code. Finally, after the code was refactored and tested, we used the MATLAB compiler to create an executable file that can be run on any Windows machine without the need of a MATLAB license, increasing the accessibility of our program.<br><br>
                            To view this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/SACNAS%20poster.pdf'>here</a>.
                                </p>
                                
                            </div><!-- panel-body -->
                        </div><!-- panel-collapse collapse-->
                    </div><!-- panel panel-default -->
                    
                </div><!-- panel-group -->
            </div>
            <div id='footer'>
                <hr id='separate'>
                <ul id='footerNav'>
                    <li><a href='index.html'>Home</a></li>
                    <li><a href='about.html'>About</a></li>
                    <li><a href='news.html'>News</a></li>
                    <li><a href='documentation.html'>Documentation</a></li>
                    <li><a href='downloads.html'>Downloads</a></li>
                    <li><a href='https://github.com/kdahlquist/GRNmap/wiki' target="_blank">Wiki</a></li>
                    <li><a href='https://github.com/kdahlquist/GRNmap/' target="_blank">Developers</a></li>
                    <li><a href='publications.html'>Publications</a></li>
                    <li><a href='people.html'>People</a></li>
                    <li><a href='contact.html'>Contact</a></li>
                    <li><a href='links.html'>Links</a></li>
                </ul>
                <p id='CCLicense'>
                    This web site is free and open to all users and there is no login requirement.
                    <br> 
                    Web site content is available under the <a href='http://creativecommons.org/licenses/by-nc-sa/3.0/' target='_blank'>Creative Commons Attribution Non-Commercial Share Alike 3.0 Unported License</a>. GRNmap code is available under the open source <a href='http://opensource.org/licenses/BSD-3-Clause' target='_blank'>BSD license</a>.
                    <br>
                    This page has been accessed <span class="ga-report"></span> times.
                </p>
            </div>
        </div>
        
        <script src="js/ga-report.js"></script>

    </body>
    
</html>
