---
layout: default
title: GRNmap - Publications
subTitle: GRNmap
lastModified: 7/12/2025/2016
---
<!--
This File can be converted to a data-driven format, low-priority
-->
<div id="accordian" class="panel-group">
    <div class="panel panel-default">
        <div class="panel-heading">
            <p class="panel-title aboutHeaders">
                <a href="#papers" data-toggle="collapse">
                    Papers
                </a>
            </p>
        </div><!-- panel-heading -->
        <div id="papers" class="panel-collapse collapse in">
            <div class="panel-body">
                <p class="abstract">
                <span style='font-weight:bold'>September 29, 2015</span><br>

                Dahlquist, K.D., Fitzpatrick, B.G., Camacho, E.T., Entzminger, S.D., and Wanner, N.C. (2015) Parameter Estimation for Gene Regulatory Networks from Microarray Data: Cold Shock Response in Saccharomyces cerevisiae. <i>Bulletin of Mathematical Biology</i>, 77(8), 1457-1492, published online September 29, 2015. DOI: 10.1007/s11538-015-0092-6.
                <br>
                <br>
                <a href="http://link.springer.com/article/10.1007/s11538-015-0092-6/fulltext.html" target="_blank">View full text in HTML </a><br>
                <a href="http://link.springer.com/content/pdf/10.1007%2Fs11538-015-0092-6.pdf" target="_blank">Download PDF (1205 KB)</a><br>
                <a href="http://link.springer.com/content/esm/art:10.1007/s11538-015-0092-6/file/MediaObjects/11538_2015_92_MOESM1_ESM.zip" target="_blank">Download electronic supplementary material (65 KB)</a><br><br>


                    <span style="font-weight:bold">Abstract</span><br>

                    We investigated the dynamics of a gene regulatory network controlling the cold shock response in budding yeast, Saccharomyces cerevisiae. The medium-scale network, derived from published genome-wide location data, consists of 21 transcription factors that regulate one another through 31 directed edges. The expression levels of the individual transcription factors were modeled using mass balance ordinary differential equations with a sigmoidal production function. Each equation includes a production rate, a degradation rate, weights that denote the magnitude and type of influence of the connected transcription factors (activation or repression), and a threshold of expression. The inverse problem of determining model parameters from observed data is our primary interest. We fit the differential equation model to published microarray data using a penalized nonlinear least squares approach. Model predictions fit the experimental data well, within the 95 % confidence interval. Tests of the model using randomized initial guesses and model-generated data also lend confidence to the fit. The results have revealed activation and repression relationships between the transcription factors. Sensitivity analysis indicates that the model is most sensitive to changes in the production rate parameters, weights, and thresholds of Yap1, Rox1, and Yap6, which form a densely connected core in the network. The modeling results newly suggest that Rap1, Fhl1, Msn4, Rph1, and Hsf1 play an important role in regulating the early response to cold shock in yeast. Our results demonstrate that estimation for a large number of parameters can be successfully performed for nonlinear dynamic gene regulatory networks using sparse, noisy microarray data.
                </p>
            </div><!-- panel-body -->
        </div><!-- panel-collapse collapse in-->
    </div><!-- panel panel-default -->

    <div class="panel panel-default">
        <div class="panel-heading">
            <p class="panel-title aboutHeaders">
                <a href="#presentations" data-toggle="collapse">
                    Presentations
                </a>
            </p>
        </div><!-- panel-heading -->
        <div id="presentations" class="panel-collapse collapse">
            <div class="panel-body">
                <p class="abstract">
                    <b><a href="https://academics.lmu.edu/media/lmuacademics/undergraduateresearchsymposium/documents/URS9_Program_LONG_with_Pages.pdf">Loyola Marymount University 9<sup>th</sup> Annual Undergraduate Research Symposium</a></b><br><br>
                    <b>March 25, 2017</b><br>
                    Systems Modeling and Statistical Analysis Allows Comparison in the Response to Cold Shock in <i>Saccharomyces cerevisiae</i> Between HAP4 and Randomly Generated Networks<br>
                    <b>Kristen M. Horstmann</b>, Ben J. Fitzpatrick, and Kam D. Dahlquist<br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) is a set of transcription factors which regulate the level of
                    expression of genes encoding other transcription factors. The dynamics of a GRN show how
                    gene expression in the network changes over time. A MATLAB software package called
                    GRNmap uses ordinary differential equations to model the dynamics of medium-scale GRNs and
                    estimates production rates, expression thresholds, and regulatory weights for each transcription
                    factor in the network based on DNA microarray data. Microarray data were obtained from a
                    Saccharomyces cerevisiae strain deleted for the Hap4 transcription factor and subjected to cold
                    shock at 13°C for 15, 30, and 60 minutes. A modified ANOVA showed that 1794 genes had a
                    log2 fold change significantly different than zero at any of the time points. These genes were
                    submitted to the YEASTRACT database to determine which transcription factors regulated them.
                    From this set, we generated a database-derived candidate GRN of 15 genes and 28 edges as well
                    as random networks of similar size. GRNmap was used to estimate the production rates,
                    expression thresholds, and regulatory weights for these networks. The Gephi software was used
                    to analyze the networks’ structures in terms of the node in- and out-degrees, eccentricity, and
                    betweenness centrality. We found that the random networks had different degree distributions
                    than the database-derived network. Also, Hap4 had a different betweenness centrality value in
                    the random networks, which affected the estimated parameter values, helping us further
                    understand its role in the cold shock response in yeast.<br><br>
                    To download the presentation slides, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/presentations/Horstmann_LMU-Symposium_20170325_talk.pptx?raw=true'>here</a>.
                 <br>
                 <br>
                 <br>
                    Comparison of the regulatory dynamics of related small gene regulatory networks that
                    control the cold shock response in <i>Saccharomyces cerevisiae</i><br>
                    <b>Natalie E. Williams</b>, Kam D. Dahlquist, Ben G. Fitzpatrick<br><br>
                    <b>Abstract</b><br>
                    The Dahlquist lab has investigated the global, transcriptional response of <i>Sacchromyces
                    cerevisiae</i>, baker’s yeast, to the environmental stress of cold shock using DNA microarrays in
                    the wild type strain and five strains deleted for a particular regulatory transcription factor. Gene
                    regulatory networks (GRNs) consist of transcription factors, genes, and the regulatory
                    connections between them that control the resulting mRNA and protein expression levels. A
                    family of six related GRNs were derived from the YEASTRACT database which ranged in size
                    from 15 to 20 genes and 27 to 36 edges. We used mathematical modeling to determine the
                    dynamics of these GRNs to determine the relative influence of each transcription factor in the
                    network. We then compared the modeling results from the database-derived network to random
                    networks with the same number of genes and edges. An initial sample of ten random networks
                    were generated. After performing parameter estimation, we found that the database-derived
                    networks performed better with smaller least-squares error values than seven of the ten random
                    networks. To perform a more robust analysis, a larger collection of random networks was
                    generated. Comparisons made between the random networks and the database-derived networks
                    consistently showed better modeling of the database-derived networks. These comparisons also
                    revealed key network motifs in both the database-derived and random networks that correlated
                    with better fits to the data. <br><br>
                    To download the presentation slides, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/presentations/LMUURS17_NEW.pptx?raw=true'>here</a>.
                </p>
                <br>
                <p class="abstract">
                    <b><a href="https://www.open-bio.org/wiki/BOSC_2016">Bioinformatics Open Source Conference (BOSC) 2016, Orlando, FL</a></b><br><br>
                    July 9, 2016<br>
                    GRNmap and GRNsight: open source software for dynamical systems modeling and visualization of medium-scale gene regulatory networks<br>
                    <b>Kam D. Dahlquist</b>, Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Trixie Anne M. Roque, Anindita Varshneya, Mihir Samdarshi, and Chukwuemeka Azinge<br><br>

                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Over a period of several years, our group has developed a MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach (Dahlquist et al. 2015, DOI: 10.1007/s11538-015-0092-6) to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. GRNmap has options for using a sigmoidal or Michaelis-Menten production function.  The large number of developers and time span of development led to a code base that was difficult to revise and adjust. We therefore brought the code under version control in a GitHub repository and refactored the script-based software with global variables into a function-based package that uses an object to carry relevant information from function to function. This modular approach allows for cleaner, less ambiguous code and increased maintainability. We standardized the format of the input and output Excel workbooks, making them more readable. We also added an optimization diagnostics output worksheet which includes both the actual and theoretical minimum least squared error overall, and the mean squared errors for the individual genes. The MATLAB compiler was used to create an executable that can be run on any Windows machine without the need of a MATLAB license, increasing the accessibility of our program. Finally, we have implemented test-driven development, creating unit tests for all new features to speed up debugging and to prevent future code regressions.  We are improving the test coverage of previous code.<br><br>
                    GRNsight is an open source web application for visualizing such models of gene regulatory networks. GRNsight accepts GRNmap- or user-generated spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph of the GRN model. It is written in JavaScript, with diagrams facilitated by D3.js. Node.js and the Express framework handle server-side functions. GRNsight’s diagrams are based on D3.js’s force graph layout algorithm, which was then extensively customized. GRNsight uses pointed and blunt arrowheads, and colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. Visualizations can be modified through manual node dragging and sliders that adjust the force graph parameters. From the early stages, GRNsight has had a unit testing framework using Mocha and the Chai assertion library to perform test-driven development where unit tests are written before new functionality is coded. This framework consists of over 160 automated unit tests that examine over 450 test files to ensure that the program is running as expected. Error and warning messages inform the user what happened, the source of the problem, and possible solutions. <br><br>
                    Together, the life cycle of these two programs illustrate the differences between the cultures of mathematics and computing, the challenges and benefits of bringing an existing code base up to open development standards (GRNmap), and the advantages of starting a project using best practices from the beginning (GRNsight).  Our goal is to facilitate reproducible research.<br><br>
                     <a href="http://f1000research.com/slides/5-1637">Click here to view the slides on F1000Research.</a><br><br>

                     <iframe src="http://www.slideshare.net/GRNsight/slideshelf" width="615px" height="470px" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:none;" allowfullscreen webkitallowfullscreen mozallowfullscreen></iframe>

                </p>
<br><br>
                <p class="abstract">
                    <span style="font-weight:bold"><a href="http://experimentalbiology.org/PDFS/2016/EB-2016-Program-Web-FullProgram.aspx">Experimental Biology 2016, San Diego, CA</a></span><br><br>
                    <b>April 4, 2016</b> <br>
                    <a href="http://www.fasebj.org/content/30/1_Supplement/819.16.abstract?sid=a80e9e6c-998f-4889-b1fd-ddcc75ab8bda">GRNmap and GRNsight: open source software for dynamical systems modeling and visualization of medium-scale gene regulatory networks</a> <br>
                    <span style="font-weight:bold">Kam D. Dahlquist</span>, Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Monica V. Hong, Kristen M. Horstmann, Kayla C. Jackson, K. Grace Johnson, Tessa A. Morris, Trixie Anne M. Roque, Mihir Samdarshi, Anindita Varshneya, Natalie E. Williams, and Kevin W. Wyllie<br><br>
                    <span style="font-weight:bold">Abstract</span><br>
                    A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Our group has developed a MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach (Dahlquist et al. 2015, DOI: 10.1007/s11538-015-0092-6) to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. GRNmap has options for using a sigmoidal or Michaelis-Menten production function. Parameters for a series of related networks, ranging in size from 15 to 35 genes, were optimized against DNA microarray data measuring the transcriptional response to cold shock in wild type and five strains individually deleted for the transcription factors, Cin5, Gln3, Hap4, Hmo1, Zap1, of budding yeast, Saccharomyces cerevisiae BY4741. Model predictions fit the experimental data well, within the 95% confidence interval. Open source code and a compiled executable that can run without a MATLAB license are available from <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>. GRNsight is an open source web application for visualizing such models of gene regulatory networks. GRNsight accepts GRNmap- or user-generated spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph of the GRN model. The application colors the edges and adjusts their thicknesses based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users can then modify the graph to define the best visual layout for the network. The GRNsight open source code and application are available from <a href="http://dondi.github.io/GRNsight/index.html">http://dondi.github.io/GRNsight/index.html</a>.
                </p>

                <figure class="publication">
                    <iframe src="//www.slideshare.net/slideshow/embed_code/key/NC3auEnK2HuL1f" width="595" height="485" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen></iframe>

                    <figcaption><a href="//www.slideshare.net/GRNsight/dahlquist-experimental-biology20160404" title="Dahlquist experimental biology_20160404" target="_blank">Presentation slides</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                </figure>
<br>

                <p class="abstract">
                    <b><a href="http://academics.lmu.edu/media/lmuacademics/undergraduateresearchsymposium/documents/ei8ht_Symposium_Long_Program_16_Secured.pdf">Loyola Marymount University 8<sup>th</sup> Annual Undergraduate Research Symposium</a></b><br><br>
                    <b>March 19, 2016</b><br>
                    Usability Improvements to GRNmap:  Software for Gene Regulatory Network Modeling and Parameter Estimation <br>
                    <b>Juan S. Carrillo Quinche</b>, <b>Trixie Anne M. Roque</b>, Kam D. Dahlquist, and John David N. Dionisio<br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) consists of a set of transcription factors that regulate the level of gene expression of other transcription factors. The dynamics of a GRN describe how gene expression in the network changes over time. GRNmap is a complex MATLAB software package that uses ordinary differential equations to model the dynamics of medium-scale GRNs, such as those from budding yeast, Saccharomyces cerevisiae. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data, and them performs forward simulations of model dynamics. Since v1.0, we have made design changes, added new features, fixed bugs, implemented a testing framework, and created documentation. Our current focus is to add functionality and make the program more user friendly. We have standardized the format of the input and output Excel workbooks, making them more readable. GRNmap can now perform multiple successive runs for fine tuning of optimization parameters. We also added an optimization diagnostics output worksheet which includes both the actual and theoretical minimum least squared error overall, and the mean squared errors for the individual genes. Using test-driven development, we created tests for all new features to speed up debugging and to prevent future code regressions. The source code and executable (which contain demo files and can run without a MATLAB license) for the updated version 1.4 are available for download at <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a> under the BSD open source license. <br><br>
                    To download the presentation slides, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/presentations/Carrillo_Roque_LMU-Symposium_20160319_talk.pptx?raw=true'>here</a>.
                 <br>
                 <br>
                 <br>
                Mathematical Modeling Shows that Gln3 Affects the Dynamics of the Gene Regulatory Network Controlling the Cold Shock Response in <i>Saccharomyces cerevisiae</i><br>
                    <b>Tessa A. Morris</b>, Kam D. Dahlquist, Ben G. Fitzpatrick<br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) consists of a set of transcription factors that regulate the expression of genes encoding other transcription factors. The focus of this study was to determine the GRN that controls the cold shock response in budding yeast, <i>Saccharomyces cerevisiae</i>, and to model its dynamics. Microarray experiments were performed in the Dahlquist lab to measure gene expression after 15, 30, and 60 minutes of cold shock for both the wild type strain and a strain deleted for the transcription factor Gln3. These data were used as input to a MATLAB software package called GRNmap, which uses ordinary differential equations to model the dynamics of a medium-scale gene regulatory network. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network using a penalized least squares approach. A modified ANOVA showed that 1356 genes (22%) had a log<sub>2</sub> fold change significantly different than zero with an adjusted p value of < 0.05 for at least one timepoint for the Gln3 deletion strain. These genes were submitted to the YEASTRACT database to determine the transcription factors that regulate them. From this, a family of 49 GRNs ranging from 35 genes and 120 edges to 14 genes and 26 edges was generated. Parameter values, production rates, regulatory weights, and expression thresholds were compared for each of these GRNs. These results show that the presence or absence of Gln3 affects the dynamics of the gene regulatory network controlling the cold shock response in yeast. <br><br>
                    To download the presentation slides, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/presentations/Morris_LMU-Symposium_20160319_talk.pptx?raw=true'>here</a>.
                </p>
                <br>
                <p class="abstract">
                    <span style="font-weight:bold"><a href="http://ccbs.uci.edu/scsb2015home">Southern California Systems Biology Conference</a></span><br><br>
                    <b>January 31, 2015</b> <br>
                    GRNmap and GRNsight: Open Source Software for Dynamical Systems Modeling and Visualization of Medium-Scale Gene Regulatory Networks<br>
                    <span style="font-weight:bold">Kam D. Dahlquist</span>,  Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Nicholas A. Rohacz, Katrina Sherbina, Britain J. Southwick, and Anindita Varshneya <br><br>
                <span style="font-weight:bold">Abstract</span><br>
                A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Our group has developed a MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. Parameters for a 21-gene network were optimized against DNA microarray data measuring the transcriptional response to cold shock in wild type and four transcription factor deletion strains of budding yeast, Saccharomyces cerevisiae. Model predictions fit experimental data well, within the 95% confidence interval. Open source code and a compiled executable that can run without a MATLAB license are available from <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>. GRNsight is an open source web application for visualizing such models of gene regulatory networks. GRNsight accepts GRNmap- or user-generated spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph of the GRN model. The application colors the edges and adjusts their thicknesses based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users can then modify the graph to define the best visual layout for the network. The GRNsight code and application are available from <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/index.html</a>. This work was partially supported by NSF award 0921038.

                </p>
                <figure class="publication">
                    <iframe src="http://www.slideshare.net/slideshow/embed_code/44236648" width="425" height="355" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen></iframe>
                    <figcaption><a href="http://www.slideshare.net/GRNsight/dahlquist-so-calsysbio20140131" title="Dahlquist so calsysbio_20140131" target="_blank">Presentation slides</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                </figure>

                </div>
            </div><!-- panel-body -->
        </div><!-- panel-collapse collapse-->
    </div><!-- panel panel-default -->

    <div class="panel panel-default">
        <div class="panel-heading">
            <p class="panel-title aboutHeaders">
                <a href="#posters" data-toggle="collapse">
                    Posters
                </a>
            </p>
        </div><!-- panel-heading -->
        <div id="posters" class="panel-collapse collapse">
            <div class="panel-body">
                <p class="abstract">
                    <b><a href="https://www.iscb.org/ismbeccb2025/home">ISMB/ECCB 2025, Liverpool, Englanda</a></b><br><br>
                    <b>July 20-25, 2025</b><br>
                    Sensitivity analysis of GRNmap and new features for GRNsight: open source software for dynamical systems modeling and visualization of small-scale gene regulatory networks<br>
                    <b>Kam D. Dahlquist</b>b, Ben G. Fitzpatrick, John David N. Dionisio, Ngoc K. Tran, Nikki C. Chun, Cecilia J. Zaragoza, A'Kaia L. Phelps, Amelie T. Dinh, Milka Y. Zekarias<br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) is a set of transcription factors that regulate the expression of genes encoding other transcription factors. GRNmap is an open source MATLAB package that uses ordinary differential equations to model dynamics of small-scale GRNs. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data and then performs forward simulations of model dynamics. While the model has been successfully used to understand networks of 15-20 genes, we wanted to closely examine how it works on a smaller scale to determine parameter sensitivity. All 21 possible 'toy' networks of 3 nodes and 4 edges were created based on the simulated expression data outputted when known parameters were run through a forward simulation. Then the simulated data was used to estimate the parameters again. Comparison of the known to estimated parameters showed that estimating production rates in addition to weights and thresholds reduced the accuracy of the results. The model was also sensitive to the direction and magnitude of the weight parameters for networks with the same connectivity. GRNsight is an open-source web application for visualizing such models of GRNs, automatically displaying a graph where edges are color-coded based on the activation and repression relationships, and nodes are color-coded with time course gene expression data. Backend gene expression and gene regulatory network databases allow users to generate a network, color the notes with expression data, and export an Excel workbook for modeling in GRNmap. <a href="http://kdahlquist.github.io/GRNmap/" http://kdahlquist.github.io/GRNmap/<a/>; <a href:="https://dondi.github.io/GRNsight/" https://dondi.github.io/GRNsight/<a/> <br>

                        <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/ISMBECCB2025_1759_Dahlquist_poster.pdf" Link to poster<a/> <br>
                        <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/ISMBECCB2025_1759_Dahlquist_poster.mp4" Link to video about poster</a> 
                <br>
                </p>
                <p class="abstract">
                    <b><a href="https://www.sccur.org/sccur/fall_2017_conference/poster_3/107/">Southern California Conference for Undergraduate Research, California State Polytechnic University, Pomona</a></b><br><br>
                    <b>November 18, 2017</b><br>
                    Restructuring the Data Architecture of GRNmap, a Gene Regulatory Network Modeling Application<br>
                    <b>Chukwuemeka E Azinge</b>, <b>Justin Kyle T. Torres</b>, John David N. Dionisio, Ben G. Fitzpatrick, and Kam D. Dahlquist<br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) consists of a set of transcription factors that regulate the level of expression of genes encoding other transcription factors. The dynamics of a GRN describe how gene expression in the network changes over time. GRNmap is a complex MATLAB software package that uses ordinary differential equations to model the dynamics of small- to medium-scale GRNs. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data, using forward simulations of model dynamics. Input is provided in the form of a multisheet Excel workbook with multiple types of data. Since our last major release, we have improved GRNmap’s usability and robustness. We changed our simple matrix into a nested cell array of matrices to handle inputs with missing values, which required a paradigm shift in our data structure. Additionally, we revisited design, implemented new features, fixed bugs, expanded the test suite, and improved documentation. We localized extraneous global variables by grouping them into a single function call to limit their scope and prevent persistence between subsequent runs, which previously led to incorrect calculations. Finally, we implemented pre-allocation of arrays which makes the program run faster as MATLAB no longer needs to calculate matrix sizes at runtime and ensures matrix operations work smoothly as this method clears previously used data to prevent persistence between runs of differently-sized networks. The open source code and executable are available for download <a href="http://kdahlquist.github.io/GRNmap/"> here </a>0under the BSD license.
                <br>
                </p>
                <p class="abstract">
                    <b><a href="https://academics.lmu.edu/media/lmuacademics/undergraduateresearchsymposium/documents/URS9_Program_LONG_with_Pages.pdf">Loyola Marymount University 9<sup>th</sup> Annual Undergraduate Research Symposium</a></b><br><br>
                    <b>March 25, 2017</b><br>
                    Dynamical Systems Modeling of Six Related Small Gene Regulatory Networks Suggest That the Transcription Factors CIN5, GLN3, HMO1, and YHP1 play a Role in Controlling the Cold Shock Response in<i>Saccharomyces cerevisiae</i><br>
                    <b>Brandon J. Klein</b>, <b>Natalie E. Williams</b>, Kam D. Dahlquist, and Ben G. Fitzpatrick<br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) is a group of transcription factors that control the level of
                    expression of genes encoding other transcription factors. Dynamics of GRNs illustrate how
                    expression in the network changes over time. GRNmap, a MATLAB software package, uses
                    differential equations to model the dynamics of medium-scale GRNs. The software estimates
                    production rates, expression thresholds, and regulatory weights for each transcription factor in
                    the network based on microarray data. Microarray data was obtained from a cold shock
                    experiment where wild type budding yeast, Saccharomyces cerevisiae, and five strains from
                    which the transcription factors Cin5, Gln3, Hap4, Hmo1, and Zap1 were deleted were subjected
                    to cold shock at 13°C for 15, 30, and 60 minutes. Six related GRNs, which ranged from 15-20
                    genes and 27-36 edges, were constructed using data from the YEASTRACT database. GRNmap
                    was then used to estimate production rates, expression thresholds, and regulatory weights for
                    each of these GRNs. Forward simulation of the model showed a good fit to the experimental
                    data, as compared to random networks with the same genes and number of edges. The
                    transcription factors Cin5, Gln3, Hmo1, and Yhp1 comprised a regulatory chain that stood out
                    because the weights were consistently conserved across five of the six GRNs. These transcription
                    factors also had among the highest total degree (in- plus out-degree) and betweenness centrality
                    values of all the genes in the networks, suggesting that they play an important role in regulating
                    the cold shock response in yeast. <br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Klein-Williams_LMU-URS-2017-Poster_20170315.pptx?raw=true'>here</a>.
                    <br>
                    <br>
                    Using Graph Statistics to Investigate the Properties of Six Candidate Gene Regulatory Networks for Controlling the Cold Shock Response in <i>Saccharomyces cerevisiae</i><br>
                    <b>Margaret J. O’Neil</b>, Kam D. Dahlquist, and Ben G. Fitzpatrick, <br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) is a set of transcription factors which regulate the level of
                    expression of genes encoding other transcription factors. The dynamics of a GRN show how
                    gene expression in the network changes over time. Microarray data were obtained from the wild
                    type strain and five transcription factor deletion strains (Δcin5, Δgln3, Δhap4, Δhmo1, Δzap1)
                    before cold shock at 30°C and after 15, 30, and 60 minutes of cold shock at 13°C. A modified
                    ANOVA showed that for all networks a large number of genes had a log2 fold change
                    significantly different than zero at any time point. These genes were submitted to the
                    YEASTRACT database to determine which transcription factors regulated them. Data from each
                    143
                    strain were used to generate six candidate GRN’s of between 14 to 17 nodes and 25 to 36 edges,
                    depending on the specific network. The open source software Gephi was used to analyze graph
                    properties of each network. In particular, we computed in- and out-degree, betweenness
                    centrality, eccentricity and closeness centrality. These centrality measures indicate which nodes
                    are most easily accessed in each network, how central a node is in a network, and which nodes
                    most frequently appear in the shortest paths of a network. From this analysis we have gained
                    insight into role of different transcription factors. In particular, the high centralities of Cin5,
                    Yhp1, and Hmo1 provide additional evidence of their potential importance in the gene regulatory
                    network that controls the cold shock response in yeast<br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/ONeil_Undergraduate_Research_Symposium_2017_poster.pptx?raw=true'>here</a>.
                    <br>
                    <br>
                    Restructuring the Data Architecture of GRNmap, a Gene Regulatory Network Modeling Application<br>
                    <b>Trixie Anne M. Roque</b>, <b>Chukwuemeka E. Azinge</b>, <b>Justin Kyle T. Torres</b>, John David N. Dionisio, Kam D. Dahlquist, and Ben G. Fitzpatrick, <br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) consists of a set of transcription factors that regulate the level
                    of expression of genes encoding other transcription factors. The dynamics of a GRN describe
                    how gene expression in the network changes over time. GRNmap is a complex MATLAB
                    software package that uses ordinary differential equations to model the dynamics of small- to
                    medium-scale GRNs. The program estimates production rates, expression thresholds, and
                    regulatory weights for each transcription factor in the network based on DNA microarray data,
                    using forward simulations of model dynamics. Since our last major release, we have focused on
                    changes that have made GRNmap a much more robust application. We revisited the design,
                    implemented new features, fixed bugs, added new cases to the test suite, and improved
                    documentation. We localized extraneous global variables by grouping them into a single function
                    call to limit their scope and prevent persistence between subsequent runs, which previously led
                    to incorrect calculations. Additionally, we changed our simple two-dimensional matrix into a
                    nested cell array of matrices to better handle inputs with missing values, which required a
                    paradigm shift in how we constructed our data structure. We also implemented pre-allocation of
                    arrays which makes the program run faster as MATLAB no longer needs to calculate matrix
                    sizes at runtime and ensures matrix operations work smoothly as this method clears previously
                    used data to prevent persistence between runs of differently-sized networks. The open source
                    code and executable are available for download at <a href='http://kdahlquist.github.io/GRNmap/'>http://kdahlquist.github.io/GRNmap/</a> under
                    the BSD license.<br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Roque_Azinge_Torres_URS-2017-poster.pdf?raw=true'>here</a>.
                </p>
                <br>
                <p class="abstract">
                    <b><a href="http://ccbs.uci.edu/scsbc2017">7<super>th</super> Annual Southern California Systems Biology Conference, University of California, Irvine</a></b><br><br>
                    <b>January 28, 2017</b><br>
                    Dynamical Systems Modeling and Gene Regulatory Network Structure Analysis Reveals HAP4's Role in Regulating the Response to Cold Shock in <i>Saccharomyces cerevisiae</i><br>
                    <b>Kristen M. Horstmann</b>,<b>Margaret J. O’Neil</b>,  Ben G. Fitzpatrick, and Kam D. Dahlquist<br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) is a set of transcription factors which regulates the level of expression of genes encoding other transcription factors. The dynamics of a GRN show how gene expression in the network changes over time. A MATLAB software package called GRNmap uses ordinary differential equations to model the dynamics of medium-scale GRNs and estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data. Microarray data were obtained from a Saccharomyces cerevisiae strain deleted for the Hap4 transcription factor subjected to cold shock at 13°C for 15, 30, and 60 minutes. A modified ANOVA showed that 1794 genes had a log2 fold change significantly different than zero at any of the time points. These genes were submitted to the YEASTRACT database to determine which transcription factors regulated them. From this we generated a candidate GRN of 15 genes and 28 edges. GRNmap was used to estimate the production rates, expression thresholds, and regulatory weights for this network. Forward simulation of the model showed a good fit with the experimental data.  The program Gephi was then used to analyze the network structure in terms of the node in- and out-degrees, eccentricity, and betweenness centrality. From this analysis we have gained further insight into Hap4's role in the gene regulatory network that controls the cold shock response in yeast.<br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Horstmann_ONeil_Southern-California-Systems-Biology-Conference_2017_poster.pptx?raw=true'>here</a>.
                <br>
                <br>
                    Dynamical Systems Modeling of Six Related Small Gene Regulatory Networks Suggest That the Transcription Factors Cin5, Hmo1, Msn2, and Yhp1 Play a Role in Controlling the Cold Shock Response in <i>Saccharomyces cerevisiae</i></a><br>
                    <b>Natalie E. Williams</b>, <b>Brandon J. Klein</b>,Ben G. Fitzpatrick, and Kam D. Dahlquist<br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) is a group of transcription factors that control the level of expression of genes encoding other transcription factors. Dynamics of GRNs illustrate how expression in the network changes over time. GRNmap, a MATLAB software package, uses differential equations to model the dynamics of medium-scale GRNs. The software estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on microarray data. Microarray data was obtained from a cold shock experiment where wild type budding yeast, Saccharomyces cerevisiae, and five strains from which the transcription factors Cin5, Gln3, Hap4, Hmo1, and Zap1 were subjected to cold shock at 13°C for 15, 30, and 60 minutes. Six related GRNs, which ranged from 15-20 genes and 27-36 edges, were constructed using data from the YEASTRACT database. GRNmap was then used to estimate production rates, expression thresholds, and regulatory weights for each of these GRNs.  Forward simulation of the model showed a good fit to the experimental data, as compared to random networks with the same genes and number of edges.  The transcription factors Cin5, Hmo1, Msn2, and Yhp1 stood out because the dynamics of their expression varied in the different deletion strains.  They also had among the highest total degree (in- plus out-degree) and betweenness centrality values of all the genes in the networks, suggesting that they play an important role in regulating the cold shock response in yeast.<br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Klein-Williams_LMU-URS-2017-Poster_20170315.pptx?raw=true'>here</a>.
                </p>
                <br>
                <p class="abstract">
                    <b><a href="https://www.open-bio.org/wiki/BOSC_2016">Bioinformatics Open Source Conference (BOSC) 2016</a> and <a href="http://www.iscb.org/ismb2016">Intelligent Systems for Molecular Biology (ISMB) 2016</a>, Orlando, FL</a></b><br><br>
                    July 8-12, 2016<br>
                    GRNmap and GRNsight: open source software for dynamical systems modeling and visualization of medium-scale gene regulatory networks<br>
                    <b>Kam D. Dahlquist</b>, Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Tessa A. Morris, Anindita Varshneya, Natalie E. Williams, K. Grace Johnson, Trixie Anne M. Roque, Kristen M. Horstmann, Mihir Samdarshi, Chukwuemeka E. Azinge, Brandon J. Klein, Margaret J. O’Neil1<br><br>

                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Our open source MATLAB software package, GRNmap (<a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>), uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach (Dahlquist et al. 2015, DOI: 10.1007/s11538-015-0092-6) to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. GRNmap has options for using a sigmoidal or Michaelis-Menten production function.  Parameters for a series of related networks, ranging in size from 15 to 35 genes, were optimized against DNA microarray data measuring the transcriptional response to cold shock in budding yeast, Saccharomyces cerevisiae, for the wild type strain and strains deleted for the transcription factors Cin5, Gln3, Hap4, Hmo1, and Zap1, giving biological insights into this process. GRNsight is an open source web application for visualizing such models of gene regulatory networks (<a href="http://dondi.github.io/GRNsight/index.html">http://dondi.github.io/GRNsight/index.html</a>). GRNsight accepts GRNmap- or user-generated Excel spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph. The application colors the edges and adjusts their thicknesses based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users can then modify the graph to define the best visual layout for the network. This work was partially supported by NSF award 0921038.<br><br>
                     <a href="http://f1000research.com/posters/5-1618">Click here</a> to view the poster on F1000Research. <a href="//www.slideshare.net/GRNsight/dahlquist-etal-boscismb2016poster-64224013" title="Dahlquist et-al bosc-ismb_2016_poster" target="_blank">Click here</a> to view poster from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare.<br><br>
                </p>
                <p class="abstract">
                    <b><a href="http://experimentalbiology.org/PDFS/2016/EB-2016-Program-Web-FullProgram.aspx">Experimental Biology 2016, San Diego, CA</a></b><br><br>
                    <b>April 4, 2016</b><br>
                    <a href="http://www.fasebj.org/content/30/1_Supplement/819.8.short">Modeling the Dynamics of a 21-gene, 50-edge Gene Regulatory Network Controlling the Transcriptional Response to Cold Shock in <i>Saccharomyces cerevisiae</i> using GRNmap</a><br>
                    <b>K. Grace Johnson</b>, Natalie E. Williams, Ben G. Fitzpatrick, and Kam D. Dahlquist<br><br>
                    <b>Abstract</b><br>
                    Gene expression is regulated by proteins called transcription factors which can either repress or activate a gene’s transcriptional output. A gene regulatory network (GRN) consists of a set of transcription factors that regulate the level of expression of genes encoding other transcription factors. The dynamics of a GRN show how gene expression in the network changes over time. Previously in the lab, a MATLAB software package called GRNmap was developed that uses ordinary differential equations to model the dynamics of medium-scale GRNs from budding yeast, <i>Saccharomyces cerevisiae</i>. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data, and then performs a forward simulation of the dynamics of the network. DNA microarray data for 6189 yeast genes was obtained from the Dahlquist lab where they subjected yeast to cold shock at 13°C and measured gene expression at three time points (after 15, 30, and 60 minutes of cold shock). We performed LOESS normalization using the limma package in R and used a modified ANOVA to determine which genes had a log2 fold change significantly different than zero at any of the timepoints studied. Using GRNmap, we estimated the parameters of a GRN derived from the YEASTRACT database, consisting of 21 nodes (transcription factors) and 50 edges (regulatory relationships). To answer our fundamental question, whether the network accurately models what actually occurs during cold shock in the yeast cell, we analyzed the results of the modeling as follows. For each gene we evaluated how well the simulated data generated by the model fit the expression profile measured by the microarrays. Factors contributing to the goodness of fit included whether the transcripton factor itself exhibited significant dynamics (in the ANOVA test) and whether the regulators of that transcription actor also showed significant dynamics. This analysis showed that, while a few genes were modeled well, others were not, potentially because they are regulated by other transcription factors that were not present in the network modeled. To investigate this further, we compared the results of the modeling of the YEASTRACT-derived network to 10 random networks that had the same nodes and and the same number of edges, but had randomized connections between nodes. We found that the random networks had larger values for the least squares error in the estimation and a different structure and degree-distribution than the YEASTRACT-derived network, suggesting that the YEASTRACT-derived network modeled the transcriptional dynamics better. To improve the performance of the model and account for missing regulators, we are now evaluating a family of YEASTRACT-derived networks by paring down an initial network of 35 nodes systematically down to 15 nodes by removing the next least significant transcription factor one by one from the network. From this analysis we expect to gain insight into the gene regulatory network that controls the cold shock response in yeast. Our working code is available on the GRNmap page (<a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>), and visualization of the network is available on GRNsight (<a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>). <br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Johnson_Williams_ONeil_LMU-Symposium_ASBMB_2016_poster.pptx?raw=true'>here</a>.
                <br>
                <br>
                    <a href="http://www.fasebj.org/content/30/1_Supplement/819.9.short">Mathematical Modeling Shows that Gln3 and Zap1 Affects the Dynamics of the Gene Regulatory Network Controlling the Cold Shock Response in <i>Saccharomyces cerevisiae</i></a><br>
                    <b>Tessa A. Morris</b>, <b>Kristen M. Horstmann</b>, <b>Kayla C. Jackson</b>, Ben G. Fitzpatrick, and Kam D. Dahlquist<br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) consists of a set of transcription factors that regulate the level of expression of genes encoding other transcription factors. The dynamics of a GRN show how gene expression in the network changes over time. While the transcriptional regulation of the response to the environmental stress of heat shock in budding yeast, <i>Saccharomyces cerevisiae</i>, is well-understood, which transcription factors regulate the early response to cold shock is not fully understood. Thus, the focus of this study was to determine the GRN that controls the cold shock response in yeast and to model its dynamics. Microarray experiments were performed in the Dahlquist lab at various timepoints after cold shock (t=15, 30, and 60 minutes) to examine the gene expression patterns of both the wild type and mutant strains of yeast that had been deleted for the transcription factors Gln3 and Zap1. The microarray data were normalized using the limma package in R and a modified ANOVA was used to determine which genes had a log2 fold change significantly different than zero at any of the timepoints studied. The genes that met the significance criterion of an adjusted Benjamini and Hochberg p < 0.05 were submitted to the YEASTRACT database to determine which transcription factors potentially regulated those genes. Two GRNs, one derived from the Gln3 deletion strain data and one from the Zap1 deletion strain data were created. The GRNs and the corresponding microarray data were then input into GRNmap, a MATLAB software package that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program estimated the production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on the microarray data, and then performed a forward simulation of the dynamics of the network. The results of the modeling for both networks were similar. The best fit of model to data was for genes directly connected to the deleted transcription factor (either Gln3 or Zap1). However, the deletion of either transcription factor from their respective networks had a large effect on the dynamics of several genes in the network that were not accounted for by the network connections. This suggests that the network structure did not fully model the actual cellular conditions of cold shock. This may be because the regulatory networks in the YEASTRACT database are based on measurements taken during other types of growth conditions, not cold shock. Our future directions include expanding the number of GRNs that we model to determine which transcription factors are missing from the current model. Our working code is available on the GRNmap page (<a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>), and visualization of the network is available on GRNsight (<a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>). <br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Morris_Horstmann_Jackson_ASBMB_2016_poster.pptx?raw=true'>here</a>.
                </p>
                <br>
                <p class="abstract">
                    <b><a href="http://academics.lmu.edu/media/lmuacademics/undergraduateresearchsymposium/documents/ei8ht_Symposium_Long_Program_16_Secured.pdf">Loyola Marymount University 8<sup>th</sup> Annual Undergraduate Research Symposium</a></b><br><br>
                    <b>March 19, 2016</b><br>
                    Mathematical Modeling Reveals Zap1’s Role in the Gene Regulatory Network that Controls the Response to Cold Shock in <i>Saccharomyces cerevisiae</i><br>
                    <b>Kristen M. Horstmann</b>, Tessa A Morris, Brandon J. Klein, Kam D. Dahlquist, and Ben G. Fitzpatrick<br><br>
                    <b>Abstract</b><br>
                    Transcription factors are proteins that act together in a gene regulatory network (GRN) by repressing or activating the expression of target genes. The purpose of this study is to determine the GRN for <i>Saccharomyces cerevisiae</i>, budding yeast, that controls the response to cold shock. The Dahlquist Lab has conducted DNA microarray experiments to measure gene expression after 15, 30, and 60 minutes of cold shock treatment for the wild type strain and a strain deleted for the Zap1 transcription factor. These data were used as input to a MATLAB software package called GRNmap, which uses ordinary differential equations to model the dynamics of a medium-scale gene regulatory network. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network using a penalized least squares approach. A modified ANOVA showed that 2559 genes (41%) had a log<sub>2</sub> fold change significantly different than zero with an adjusted p value of < 0.05 for at least one timepoint for the Zap1 deletion strain. These genes were submitted to the YEASTRACT database to determine the transcription factors that regulate them. From this, a family of GRNs ranging from 34 genes and 98 edges to 15 genes and 27 edges was generated. Parameter values, production rates, regulatory weights, and expression thresholds were compared for each of these GRNs. From the modeling of the network families, we have observed that Zap1 plays an important role in the gene regulatory network that controls cold shock response in <i>Saccharomyces cerevisiae</i>. <br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Horstmann_Morris_Klein_LMU-Symposium_2016_poster.pptx?raw=true'>here</a>.
                 <br>
                 <br>
                Evaluating Hap4’s Role in the Gene Regulatory Network that Controls the Response to Cold Shock in <i>Saccharomyces cerevisiae</i> using GRNmap<br>
                    <b>K. Grace Johnson</b>, <b>Margaret J. O’Neil</b>, Kam D. Dahlquist, and Ben G. Fitzpatrick, <br><br>
                    <b>Abstract</b><br>
                    Gene expression is regulated by proteins called transcription factors which either repress or activate a gene’s transcriptional output. A gene regulatory network (GRN) is a set of transcription factors that regulate the level of expression of genes encoding other transcription factors. The dynamics of a GRN show how gene expression in the network changes over time. A MATLAB software package called GRNmap uses ordinary differential equations to model the dynamics of medium-scale GRNs from budding yeast, <i></i>Saccharomyces cerevisiae</i>. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data. Data were obtained from a yeast strain deleted for the Hap4 transcription factor subjected to cold shock at 13°C for 15, 30, and 60 minutes. A modified ANOVA showed that 1794 genes had a log<sub>2</sub> fold change significantly different than zero at any of the timepoints. These genes were submitted to the YEASTRACT database to determine their transcription factors regulators. From this we generated 32 candidate GRNs that ranged in size from 35 genes, 102 edges to 15 genes, 28 edges. We then estimated and compared the parameter values for production rates, expression thresholds, and regulatory weights for each GRN. A comparison of the actual least squares error to the minimum theoretical least squares error allowed us to evaluate which size network best explains gene expression. From this analysis we have gained insight into Hap4's role in the gene regulatory network that controls the cold shock response in yeast. <br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Johnson_Williams_ONeil_LMU-Symposium_ASBMB_2016_poster.pptx?raw=true'>here</a>.
                </p>
                <br>
                <p class='abstract'>
                    <!--font-weight:bold vs. strong vs. b... idc what you pick but be consistent dang it! -->
                    <span style='font-weight:bold'><a href="http://www.2015sacnas.org/events/2015-sacnas-national-conference/event-summary-6170a953907b4808ab42c5ec7ec147b6.aspx">SACNAS (Society for the Advancement of Chicanos and Native Americans in Science) National Conference</a></span><br><br>
                    <b>October 29-31, 2015</b><br>
                    Test-Driven Development and Functionality Improvements to GRNmap, a Gene Regulatory Network Modeling Application <br>
                    <b>Trixie Anne M. Roque</b>, Tessa A. Morris, Kam D. Dahlquist, John David N. Dionisio, Ben G. Fitzpatrick <br><br>
                    <span style="font-weight:bold">Abstract</span><br>
                    A gene regulatory network (GRN) consists of a set of transcription factors that regulate the level of gene expression encoding other transcription factors. The dynamics of a GRN is how gene expression in the network changes over time. GRNmap is a complex MATLAB software package that uses ordinary differential equations to model the dynamics of medium-scale GRNs from budding yeast, <i>Saccharomyces cerevisiae</i>. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data, and then performs a forward simulation of the network dynamics. Since v1.0, we have made design changes, added new features, fixed bugs, implemented a testing a framework, and created documentation. For example, GRNmap now accepts and outputs Excel worksheets with more descriptive names, computes the standard deviations of the log2 expression data, and outputs an optimization diagnostics sheet which includes both actual and minimum least squares error. We have also designed sixteen manual input sheet tests to uncover and fix bugs. We incorporated these tests into an automated testing framework that will speed debugging and prevent future code regressions. We have added documentation to our website and wiki and constructed a UML activity diagram to document the program's overall flow and how each function processes information. The source code and executable (which contain demo files and can run without a MATLAB license) for the updated version 1.2 are available for download at <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a> under the BSD open source license. <!--note that I'm not sure if this is the final version of the abstract because Trixie noted that Dr. Fitzpatrick had also made changes to the version she sent--> <br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Roque_McNair_SACNAS_2015_poster.pdf?raw=true'>here</a>.
                </p>
                <br>
                <p class='abstract'>
                    <b><a href="http://www.pointloma.edu/experience/academics/schools-departments/department-biology/science-education-programs/wcbsurc/40th-annual-wcbsurc-plnu-2015">40<sup>th</sup> Annual West Coast Biological Sciences Undergraduate Research Conference</a></b><br><br>
                    <b>April 25, 2015</b><br>
                    Comparing the Dynamics of the Cold Shock Gene Regulatory Network in Yeast with a Random Network<br>
                    <b>Natalie E. Williams</b>, <b>K. Grace Johnson</b>, Kam D. Dahlquist, Ben G. Fitzpatrick <br><br>
                    <span style="font-weight:bold">Abstract</span><br>
                    A gene regulatory network (GRN) consists of a set transcription factors that regulate the level of expression of genes encoding other transcription factors. The dynamics of a GRN is how gene expression in the network changes over time. Previously in the lab, a mathematical model called GRNmap was developed that uses ordinary differential equations to model the dynamics of medium-scale GRNs from yeast. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data, and then performs a forward simulation of the dynamics of the network. We performed statistical analysis of DNA microarray data generated in the Dahlquist Lab for the wild type strain and four transcription factor deletion strains. These data were used to define a GRN of 21 transcription factors that are thought to be involved in regulating the transcriptional response to cold shock in yeast. To validate the model, we then compared this network to ten different networks that had the same nodes and the same number of edges, but varying connectivity. We found that the degree distribution of the experimentally-derived network was different than the random networks. We also found that the parameter values, least square error, and penalty terms varied between the experimentally-derived and random networks. We are currently analyzing this information to develop a method to use the results of the dynamical model to help us decide which gene regulatory network better models what is going on in the cell. <br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Johnson_Williams_LMU-Symposium_2015_poster.pptx?raw=true'>here</a>.
                </p>
                <br>

                <p class='abstract'>
                    <span style="font-weight:bold"><a href="http://academics.lmu.edu/media/lmuacademics/undergraduateresearchsymposium/documents/Final%20Program%20w.%20Abstracts.pdf">Loyola Marymount University 7<sup>th</sup> Annual Undergraduate Research Symposium</a></span><br><br>
                    <b>March 21, 2015</b> <br>
                    Software Refactoring and Usability Enhancement for GRNmap, a Gene Regulatory Network Modeling Application<br>
                    <b>Juan Carrillo</b>, <b>Trixie Anne Roque</b>, Kam D. Dahlquist, Ben G. Fitzpatrick <br><br>
                    <span style="font-weight:bold">Abstract</span><br>
                    A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. The dynamics of a GRN is how gene expression in the network changes over time. Over a period of several years, our group has developed a complex MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs from budding yeast, Saccharomyces cerevisiae. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data, and then performs a forward simulation of the dynamics of the network. The large number of developers and time span of development led to a code base that was difficult to revise and adjust. We therefore refactored the script-based software with global variables into a function-based package that uses an object to carry relevant information from function to function. This modular approach allows for cleaner, less ambiguous code and increased maintainability. We then used the MATLAB compiler to create an executable file that can be run on any Windows machine without the need of a MATLAB license, increasing the accessibility of our program. The GRNmap code and executable is available from <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a> under an open source license. We have now adopted a test-driven development framework that will speed debugging and ensure that future modifications to the code preserve correct functionality.<br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Carrillo_Roque_LMU-Symposium_2015_poster.pptx?raw=true'>here</a>.
                <br>
                <br>
                    Comparing the Dynamics of the Cold Shock Gene Regulatory Network in Yeast with a Random Network<br>
                    <b>Natalie E. Williams</b>, <b>K. Grace Johnson</b>, Kam D. Dahlquist, Ben G. Fitzpatrick <br><br>
                    <span style="font-weight:bold">Abstract</span><br>
                    A gene regulatory network (GRN) consists of a set transcription factors that regulate the level of expression of genes encoding other transcription factors. The dynamics of a GRN is how gene expression in the network changes over time. Previously in the lab, a mathematical model called GRNmap was developed that uses ordinary differential equations to model the dynamics of medium-scale GRNs from yeast. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data, and then performs a forward simulation of the dynamics of the network. We performed statistical analysis of DNA microarray data generated in the Dahlquist Lab for the wild type strain and four transcription factor deletion strains. These data were used to define a GRN of 21 transcription factors that are thought to be involved in regulating the transcriptional response to cold shock in yeast. To validate the model, we then compared this network to ten different networks that had the same nodes and the same number of edges, but varying connectivity. We found that the degree distribution of the experimentally-derived network was different than the random networks. We also found that the parameter values, least square error, and penalty terms varied between the experimentally-derived and random networks. We are currently analyzing this information to develop a method to use the results of the dynamical model to help us decide which gene regulatory network better models what is going on in the cell. <br><br>
                    To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Johnson_Williams_LMU-Symposium_2015_poster.pptx?raw=true'>here</a>.
                </p>
                <br>

                <p class='abstract'>
                    <span style='font-weight:bold'><a href ='http://sacnas.org/events/national-conf/past/2014'>SACNAS (Society for the Advancement of Chicanos and Native Americans in Science) National Conference</a></span> <br><br>
                    <b>October 16-18, 2014</b><br>
                    Software Refactoring and Usability Enhancement for GRNmap, a Gene Regulatory Network Modeling Application <br>
                    <b>Juan Carrillo</b>, Katrina Sherbina, Kam D. Dahlquist, Ben G. Fitzpatrick <br><br>
                    <span style="font-weight:bold">Abstract</span><br>
                    A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. The dynamics of a GRN is how gene expression in the network changes over time. Over a period of several years, our group has developed a complex MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of a 21-transcription factor GRN from budding yeast, Saccharomyces cerevisiae. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on DNA microarray data, and then performs a forward simulation of the dynamics of the network. The large number of developers and time span of development led to a code base that is difficult to revise and adjust. We therefore refactored the script-based software with global variables into a function-based package that uses an object to carry relevant information from function to function. This modular approach allows for cleaner, less ambiguous code and increased maintainability. Further revisions to the model will also be easier to implement. In addition, we have added a simple user interface, removing the need for users to edit MATLAB code. Finally, after the code was refactored and tested, we used the MATLAB compiler to create an executable file that can be run on any Windows machine without the need of a MATLAB license, increasing the accessibility of our program.<br><br>
            To download this poster, click <a href='https://github.com/kdahlquist/GRNmap/blob/master/documents/posters/Carrillo_SACNAS_2014_poster.pdf?raw=true'>here</a>.
                </p>

            </div><!-- panel-body -->
        </div><!-- panel-collapse collapse-->
    </div><!-- panel panel-default -->

</div><!-- panel-group -->
